Assets/Scripts/Controllers/FurnitureSpriteController.cs

```diff
@@ -115,19 +113,40 @@ public Sprite GetSpriteForFurniture(Furniture furn)
         // Otherwise, the sprite name is more complicated.
         spriteName += "_";
 
-        // Check for neighbours North, East, South, West.
+        // Check for neighbours North, Northeast, East, Southeast, South, Southwest, West, Northwest.
         int x = furn.Tile.X;
         int y = furn.Tile.Y;
 
-        spriteName += GetSuffixForNeighbour(furn, x, y + 1, "N");
-        spriteName += GetSuffixForNeighbour(furn, x + 1, y, "E");
-        spriteName += GetSuffixForNeighbour(furn, x, y - 1, "S");
-        spriteName += GetSuffixForNeighbour(furn, x - 1, y, "W");
+        string suffix = string.Empty;
 
-        // For example, if this object has all four neighbours of
+        suffix += GetSuffixForNeighbour(furn, x, y + 1, "N");
+        suffix += GetSuffixForNeighbour(furn, x + 1, y, "E");
+        suffix += GetSuffixForNeighbour(furn, x, y - 1, "S");
+        suffix += GetSuffixForNeighbour(furn, x - 1, y, "W");
+        if (suffix.Contains("N") && suffix.Contains("E"))
```

Comment:
I don't think you realise how `GetSuffixForNeighbour` works: It checks the tile at the given xy-Coordinates to see if it has exactly that furniture. If I did this code now, I'd be checking for the tile the furniture asking if it has neighbours is on, i. e. the same furniture I hand the method over as `furn`. Believe me, this would not work.